### ISSUE TITLE
Seccomp filter breakage with glibc-2.38 on host and glibc-2.39 in nspawn

### ISSUE THREAD

**Issue by akhuettel** [systemd/systemd#31469]
### systemd version the issue has been seen with

255.3

### Used distribution

Gentoo

### Linux kernel version used

6.6.13-gentoo-dist

### CPU architectures issue was seen on

x86_64

### Component

systemd-nspawn

### Unexpected behaviour you saw

Host has glibc-2.38, in a nspawn glibc-2.39 is installed. 
Unpacking a tar file as root in the nspawn leads to a lot of permission errors (while it should just silently succeed).
Starting systemd-nspawn with SYSTEMD_SECCOMP=0 restores normal operation.

### Steps to reproduce the problem

```
milou /var/lib/machines # systemd-nspawn -M dilfridge-amd64
Spawning container dilfridge-amd64 on /var/lib/machines/dilfridge-amd64.
Press Ctrl-] three times within 1s to kill container.
dilfridge-amd64 ~ # tar xfj portage-3.0.62.tar.bz2 
tar: portage-3.0.62/.builds: Cannot change mode to rwxrwxr-x: Operation not permitted
tar: portage-3.0.62/.github/workflows: Cannot change mode to rwxrwxr-x: Operation not permitted
tar: portage-3.0.62/.github: Cannot change mode to rwxrwxr-x: Operation not permitted
tar: portage-3.0.62/bin/chmod-lite: Cannot change mode to rwxrwxrwx: Operation not permitted
tar: portage-3.0.62/bin/ebuild-helpers/bsd: Cannot change mode to rwxrwxr-x: Operation not permitted
tar: portage-3.0.62/bin/ebuild-helpers/eerror: Cannot change mode to rwxrwxrwx: Operation not permitted
tar: portage-3.0.62/bin/ebuild-helpers/einfo: Cannot change mode to rwxrwxrwx: Operation not permitted
tar: portage-3.0.62/bin/ebuild-helpers/eqawarn: Cannot change mode to rwxrwxrwx: Operation not permitted
tar: portage-3.0.62/bin/ebuild-helpers/ewarn: Cannot change mode to rwxrwxrwx: Operation not permitted
...
```

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
Feb 23 22:01:42 milou systemd-machined[922777]: New machine dilfridge-amd64.
Feb 23 22:01:42 milou systemd[1]: Started machine-dilfridge\x2damd64.scope.
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: kauditd_printk_skb: 16 callbacks suppressed
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:28): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:29): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:30): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:31): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:32): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:33): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:34): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:35): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:36): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou kernel: audit: type=1326 audit(1708725711.023:37): auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
Feb 23 22:01:51 milou audit[1888682]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=1888682 comm="tar" exe="/usr/bin/gtar" sig=0 arch=c000003e syscall=452 compat=0 ip=0x7fd8f795dc7e code=0x50000
```

**Comment by akhuettel** [comment]
Note the systemd version on the host is 254.8 which may explain things.

**Comment by yuwata** [comment]
I guess, this is a duplicate of #30250, and already fixed by #30291 and #30305.

**Comment by akhuettel** [comment]
At least on my test machine resolved by updating the host to 255.3 (the original reporter is asleep now). Most likely already fixed indeed.
